apps/rush-lib/src/logic/setup/TerminalInput.ts

```diff
@@ -0,0 +1,227 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as readline from 'readline';
+import * as process from 'process';
+import colors from 'colors';
+
+import { KeyboardLoop } from './KeyboardLoop';
+import { AnsiEscape } from '@rushstack/node-core-library';
+
+export interface IBasePromptOptions {
+  question: string;
+}
+
+export interface IPromptYesNoOptions extends IBasePromptOptions {
+  defaultValue?: boolean | undefined;
+}
+
+export interface IPromptPasswordOptions extends IBasePromptOptions {
+  /**
+   * The string length must not be longer than 1.  An empty string means to show the input text.
+   * @defaultValue `*`
+   */
+  passwordCharacter?: string;
+}
+
+export interface IPromptLineOptions extends IBasePromptOptions {}
+
+class YesNoKeyboardLoop extends KeyboardLoop {
+  public readonly options: IPromptYesNoOptions;
+  public result: boolean | undefined = undefined;
+
+  public constructor(options: IPromptYesNoOptions) {
+    super();
+    this.options = options;
+  }
+
+  protected onStart(): void {
+    this.stderr.write(colors.green('==>') + ' ');
+    this.stderr.write(colors.bold(this.options.question));
```

Comment:
Why `stderr`?

Reply:
I don't think TerminalInput would ever work with redirected `stdin`.  It requires a live TTY.  I didn't consider or test this scenario in depth, yet.